### PR TITLE
[#91] [Test] As a developer, I can run automated UI tests for home screen

### PR DESCRIPTION
**Pull request by Shayokh144**
#91 

## What happened 👀

- Add UI test for swipe survey screen
- Add UI test for survey details button
- Add UI test for start survey button

## Insight 📝

- Add Ui test for home screen to start survey flow screen.
- Use ValueKey to find widgets

## Proof Of Work 📹

https://github.com/nimblehq/ic-flutter-taher-toby/assets/13636465/255ba7ee-8af8-4ab5-906a-c8fcd01dcef8

### CODE REVIEW

**Comment by Tuubz** [comment]
@luongvo Two final (small) PRs ready for your review 😍

File: integration_test/app_test.dart [hunk 1]
```diff
@@ -63,4 +64,29 @@ void main() async {
...
+    ));
+    await tester.pumpAndSettle();
+    String homeScreenTextValue = 'Today';
+    expect(find.text(homeScreenTextValue), findsOneWidget);
+
+    await tester.flingFrom(const Offset(100, 300), const Offset(-100, 0), 500);
+    await tester.pumpAndSettle();
+
+    String surveyTitleText = 'ibis Bangkok Riverside';
+    expect(find.text(surveyTitleText), findsOneWidget);
```
[the remark this answers is not in the document]
Comment (Tuubz) [reply integration_test/app_test.dart:80]: Yes, we intentionally assert the second survey page so we can prove that the swipe navigation works for the view pager 🙏

[the remark this answers is not in the document]
Comment (luongvo) [reply integration_test/app_test.dart:80]: Should we also assert the first page content?

File: integration_test/app_test.dart [hunk 2]
```diff
@@ -63,4 +64,29 @@ void main() async {
     String homeScreenTextValue = 'Today';
     expect(find.text(homeScreenTextValue), findsOneWidget);
   });
+
+  testWidgets('Home screen test', (WidgetTester tester) async {
+    await tester.pumpWidget(const ProviderScope(
+      child: SurveyApp(),
+    ));
```
[the remark this answers is not in the document]
Comment (Tuubz) [reply integration_test/app_test.dart:71]: From my understanding, the CI will always start with the login success test first

But now you mention it, we forgot to think about the case when the user runs this test individually 💭 

cc @Shayokh144

File: integration_test/app_test.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (Tuubz) [reply integration_test/app_test.dart:80]: Fine for me 🙏 

cc @Shayokh144

**Review by luongvo — APPROVED** [review]
This could be a way to bypass the logged in state. However, there is something we can optimize to `only have actions to login and navigate` to the Survey form from the first survey page instead of repeating the asserting steps 💭 
- _loginIn()
- _openSurveyForm()

**Pull request merged**